### PR TITLE
Dropdown, Modal, Popover nesting

### LINKED ISSUE #850: Use much complex solution for Modals, Dropdowns, Popovers
Use much complex solution for Modals, Dropdowns, Popovers to prevent bugs with overflow: hidden.

We need to make a fixed dropdown, that can be shown outside overflow-hidden card and divs.

There is some example: https://codepen.io/agop/pen/itbew

### PR DESCRIPTION
**Pull request by DieselTripleX**
## Description

closes #850

First of all, merged #859 and #880.
1. Added teleport functional for dropdowns and modals to prevent overflowing.
  1.1 Added <attachElement> prop that responsible for teleport destination selector (Default: "body").
  1.2 Added <disableAttachment> prop that disables teleport feature (In rare cases, might be useful).
2. Removed popover component from the project.
3. Fixed select component styling issue, after teleporting dropdown to DOM root.
4. Changed demo's descripition for incoming fixes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/vuestic-components/va-modal/VaModal.vue [hunk 1]
```diff
@@ -1,82 +1,84 @@
 <template>
```
Comment (asvae) [inline packages/ui/src/components/vuestic-components/va-modal/VaModal.vue:1]: Nested modal doesn't work correctly.

File: packages/ui/src/components/vuestic-components/va-modal/VaModal.vue [hunk 2]
```diff
@@ -92,6 +94,8 @@ import VaIcon from '../va-icon'
 
 class ModalProps {
   modelValue = prop<boolean>({ type: Boolean, default: false })
+  attachElement = prop<string>({ type: String, default: 'body' })
```
Comment (asvae) [inline packages/ui/src/components/vuestic-components/va-modal/VaModal.vue:97]: can we move this to composable file?

**Comment by asvae** [comment]
Feel free to merge with squash after fix.

**Review by asvae — APPROVED** [review]

File: packages/ui/src/components/vuestic-components/va-modal/VaModal.vue (hunk 1 above)
Comment (DieselTripleX) [reply packages/ui/src/components/vuestic-components/va-modal/VaModal.vue:1]: 👌 Got it!

**Review by m0ksem — CHANGES_REQUESTED** [review]
Modal can not be closed. It causes an error inside Vue and then breaks vue-i18n.

image

**Comment by m0ksem** [comment]
Remade modals a little bit, nested modals work fine now.

**Comment by m0ksem** [comment]
Docs required here. Also, you don't specify attachment props for VaButtonDropdown and other components that using VaDropdown.

**Comment by m0ksem** [comment]
I think we can provide `attachElement` prop using VaConfig in some rare cases, so we don't need a special prop to pass to VaDropdown. 

I suppose we can make some service that will allow us to pass props to specific child components later.
For example, we can create a config directive
```vue
<va-select ... v-components-config="{ VaDropdown: { 'close-on-click-outside': false } }" />
```

For now solution will be
```vue
<va-config :components="{ VaDropdown: { 'close-on-click-outside': false } }">
  <va-select ... />
</va-config>
```

Anyway passing props to VaDropdown in some components can be a rare case so we can handle it later.

**Comment by asvae** [comment]
Thanks!

**Pull request merged**